syft/workers/plan.py

```diff
@@ -0,0 +1,261 @@
+
+        Then if it has a remote location, send the plan to the remote location only the
+        first time, request a remote plan execution with specific pointers and ids for
+        storing the result, and return a pointer to the result of the execution.
+
+        If the plan is local: update the plan with the result_ids and args ids given,
+        run the plan and return the None message serialized.
+        """
+        first_run = self.readable_plan == []
+        if first_run:
+            self.build_plan(args)
+
+        if self.location:
+            if self.ptr_plan is None:
+                self.ptr_plan = self._send(self.location)
+            response = self.request_execute_plan(result_ids, *args)
+            return response
+
+        # if the plan is not to be sent but is not local (ie owned by the local worker)
+        if not self.location and self.owner != sy.hook.local_worker:
+            arg_ids = [arg.id for arg in args]
+            self.replace_ids(self.arg_ids, arg_ids)
+            self.arg_ids = arg_ids
+
+            self.replace_ids(self.result_ids, result_ids)
+            self.result_ids = result_ids
+
+            for message in self.readable_plan:
+                bin_message = sy.serde.serialize(message, simplified=True)
+                self.owner.recv_msg(bin_message)
+
+        return sy.serde.serialize(None)
+
+    def request_execute_plan(self, response_ids, *args, **kwargs):
+        """
+        Send a request to execute the plan on the remote location
+        :param response_ids: where the plan result should be stored remotely
+        :param args: the arguments use as input data for the plan
+        :return:
+        """
+        args = [args, response_ids]
+        command = ("execute_plan", self.ptr_plan, args, kwargs)
+
+        response = self.owner.send_command(
+            message=command, recipient=self.location, return_ids=response_ids
+        )
+        return response
+
+    def create_pointer(
+        self,
+        location: BaseWorker = None,
+        id_at_location: (str or int) = None,
+        register: bool = False,
+        owner: BaseWorker = None,
+        ptr_id: (str or int) = None,
+    ) -> PlanPointer:
+        return PlanPointer(ptr_id, location, id_at_location, owner)
+
+    def send(self, location):
+        """
+        Mock send function that only specify that the Plan will have to be sent to location
+        """
+        if self.location is not None:
+            raise NotImplementedError(
+                "Can't send a Plan which already has a location, use .get() before"
+            )
+        else:
+            self.location = location
+        return self
+
+    def get(self):
+        """
+        Mock get function
+        """
+        self.replace_worker_ids(self.location.id, self.owner.id)
+        self.location = None
+        self.ptr_plan = None
+        return self
+
+    def _send(self, location):
```

Comment:
Why is the real send private but the mock send public?

Reply:
The interest is that you can call .send() but actually this doesn't trigger a call to a remote worker, just "a promise" that it will be sent (with _send() ) as soon the plan in called (and built)